### PR TITLE
feat: Add measurement widget

### LINKED ISSUE #258: PLOTTING: measure widget
Add feature for measuring the distance between two points within PyVista visualization.

This is dependent upon PyVista implementation. Here is the backlog item:
https://github.com/pyvista/pyvista/issues/588

Once it is merged within PyVista main, can pull it into PyGeometry

### PR DESCRIPTION
**Pull request by AlejandroFernandezLuces**
Closes  #258

### CODE REVIEW

**Review by RobPasMue — COMMENTED** [review]
Is it ready for review? =)

**Comment by AlejandroFernandezLuces** [comment]
> Is it ready for review? =)

Not yet, I'll let you know when it's ready

**Comment by AlejandroFernandezLuces** [comment]
This is the current status of the measurement widget, from what I've seen in https://github.com/pyvista/pyvista/pull/3718/files , the widget is supposed to be snapping to the points of the mesh, but it is not working from PyVista side.

Also, I'm using calls to VTK directly to turn the widget on and off, since using the [PyVista expected method for this](https://docs.pyvista.org/version/stable/api/plotting/_autosummary/pyvista.plotting.widgets.WidgetHelper.clear_measure_widgets.html) seems to be interfering with the picker.


Recording 2023-09-15 at 13 31 08

**Comment by AlejandroFernandezLuces** [comment]
I managed to fix the snapping issue, I had to do way more changes for this to work than I expected. 

Recording 2023-09-15 at 15 59 49

File: src/ansys/geometry/core/plotting/plotter_helper.py [hunk 1]
```diff
@@ -204,6 +226,16 @@ def compute_edge_object_map(self) -> Dict[pv.Actor, EdgePlot]:
             for edge in object.edges:
                 self._edge_actors_map[edge.actor] = edge
 
+    def enable_picking(self):
+        """Enable picking capabilities in the plotter."""
+        self._pl.scene.enable_mesh_picking(
+            callback=self.picker_callback, use_actor=True, show=False, show_message=False
+        )
+
+    def disable_picking(self):
```
Comment (RobPasMue) [inline src/ansys/geometry/core/plotting/plotter_helper.py:235]: It seems that enabling picking is activated by default... how do you disable it then?

Comment (RobPasMue) [reply src/ansys/geometry/core/plotting/plotter_helper.py:235]: I can see the method, but no actual call to it.

Comment (AlejandroFernandezLuces) [reply src/ansys/geometry/core/plotting/plotter_helper.py:235]: It is called when activating the measurement widget, so it doesn't interfere with each other. 

The function is called here: https://github.com/ansys/pyansys-geometry/blob/c7b52122e06b512c58ed727a13e38e3ca202a0bf/src/ansys/geometry/core/plotting/widgets/measure.py#L71-L74

Regarding the first comment, I forgot to use the `allow_picking` flag to the proper places, already fixed 🙂

File: src/ansys/geometry/core/plotting/plotter_helper.py [hunk 2]
```diff
@@ -68,7 +77,7 @@ def __init__(
         self._picked_list = set()
         self._picker_added_actors_map = {}
         self._edge_actors_map = {}
-
+        self._widgets = []
         if self._use_trame and _HAS_TRAME:
```
Comment (RobPasMue) [inline src/ansys/geometry/core/plotting/plotter_helper.py:81]:
```suggestion
        self._widgets = []

        if self._use_trame and _HAS_TRAME:
```

**Review by RobPasMue — DISMISSED** [review]
Great job @AlejandroFernandezLuces - Update the cache and let's merge it.

Unless you prefer to do the increased transparency of the black shapes here.

**Review by RobPasMue — COMMENTED** [review]
Looks good! Let's bump the cache

**Review by RobPasMue — APPROVED** [review]

**Pull request merged**